internal/scaffold/role.go

```diff
@@ -143,6 +145,111 @@ func UpdateRoleForResource(r *Resource, absProjectPath string) error {
 	}
 }
 
+// MergeRoleForResource merges incoming new API resource rules with existing deploy/role.yaml
+func MergeRoleForResource(r *Resource, absProjectPath string, roleScaffold Role) error {
+	roleFilePath := filepath.Join(absProjectPath, DeployDir, RoleYamlFile)
+	roleYAML, err := ioutil.ReadFile(roleFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read role manifest %v: %v", roleFilePath, err)
+	}
+	// Check for existing role.yaml
+	if len(roleYAML) == 0 {
+		return fmt.Errorf("empty Role File at: %v", absProjectPath)
+	}
+	// Check for incoming new Role
+	if len(roleScaffold.CustomRules) == 0 {
+		return fmt.Errorf("customRules cannot be empty for new Role at: %v", r.APIVersion)
+	}
+
+	obj, _, err := cgoscheme.Codecs.UniversalDeserializer().Decode(roleYAML, nil, nil)
+	if err != nil {
+		return fmt.Errorf("failed to decode role manifest %v: %v", roleFilePath, err)
+	}
+	switch role := obj.(type) {
+	case *rbacv1.Role:
+		// TODO: Add logic to merge Cluster scoped rules into existing Kind: Role scoped rules
+		// Error out for ClusterRole merging with existing Kind: Role
+		if roleScaffold.IsClusterScoped {
+			return fmt.Errorf("cannot Merge Cluster scoped rules with existing deploy/role.yaml. " +
+				"please modify existing deploy/role.yaml and deploy/role_binding.yaml " +
+				"to reflect Cluster scope and try again")
+		}
+		mergedRoleRules := mergeRules(role.Rules, roleScaffold)
+		role.Rules = mergedRoleRules
+
+		if err := updateRoleFile(&role, roleFilePath); err != nil {
+			return fmt.Errorf("failed to update for resource (%v, %v): %v",
+				r.APIVersion, r.Kind, err)
+		}
+	case *rbacv1.ClusterRole:
+		mergedClusterRoleRules := mergeRules(role.Rules, roleScaffold)
+		role.Rules = mergedClusterRoleRules
+
+		if err := updateRoleFile(&role, roleFilePath); err != nil {
+			return fmt.Errorf("failed to update for resource (%v, %v): %v",
+				r.APIVersion, r.Kind, err)
+		}
+	default:
+		log.Errorf("Failed to parse role.yaml as a role %v", err)
+	}
+
+	return UpdateRoleForResource(r, absProjectPath)
+}
+
+func ifMatches(pr []string, newPr []string) bool {
+
+	sort.Strings(pr)
+	sort.Strings(newPr)
+
+	if len(pr) != len(newPr) {
+		return false
+	}
+	for i, v := range pr {
+		if v != newPr[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func findResource(resources []string, search string) bool {
+	for _, r := range resources {
+		if r == search || r == "*" {
+			return true
+		}
+	}
+	return false
+}
+
+func mergeRules(rules []rbacv1.PolicyRule, roleScaffold Role) []rbacv1.PolicyRule {
```

Comment:
Generalize the parameters:
```suggestion
func mergeRules(rules1, rules2 []rbacv1.PolicyRule) []rbacv1.PolicyRule {
```